hypershift-operator/controllers/scheduler/azure/controller.go

```diff
@@ -0,0 +1,148 @@
+	return builder.Complete(r)
+}
+
+func (r *Scheduler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	hc := &hyperv1.HostedCluster{}
+	log := ctrl.LoggerFrom(ctx)
+	err := r.Get(ctx, req.NamespacedName, hc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("hostedcluster not found, aborting reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
+	}
+	if !hc.DeletionTimestamp.IsZero() {
+		log.Info("hostedcluster is deleted, cleaning up")
+		if controllerutil.ContainsFinalizer(hc, schedulerFinalizer) {
+			if controllerutil.ContainsFinalizer(hc, hostedcluster.HostedClusterFinalizer) {
+				// Wait until the hosted cluster finalizer is removed
+				return ctrl.Result{}, nil
+			}
+			// Ensure that any placeholder deployment is deleted
+			controllerutil.RemoveFinalizer(hc, schedulerFinalizer)
+			if err := r.Update(ctx, hc); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		return ctrl.Result{}, nil
+	}
+	if !controllerutil.ContainsFinalizer(hc, schedulerFinalizer) {
+		controllerutil.AddFinalizer(hc, schedulerFinalizer)
+		if err := r.Update(ctx, hc); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	isPaused, duration, err := util.ProcessPausedUntilField(hc.Spec.PausedUntil, time.Now())
+	if err != nil {
+		log.Error(err, "error processing hosted cluster paused field")
+		return ctrl.Result{}, nil // user needs to reformat the field, returning error is useless
+	}
+	if isPaused {
+		log.Info("Reconciliation paused", "pausedUntil", *hc.Spec.PausedUntil)
+		return ctrl.Result{RequeueAfter: duration}, nil
+	}
+
+	desiredSize := hc.Labels[hyperv1.HostedClusterSizeLabel]
+	if desiredSize == "" {
+		log.Info("HostedCluster does not have a size label, skipping for now")
+		return ctrl.Result{}, nil
+	}
+	log = log.WithValues("desiredSize", desiredSize)
+
+	config := schedulingv1alpha1.ClusterSizingConfiguration{}
+	if err := r.Get(ctx, types.NamespacedName{Name: "cluster"}, &config); err != nil {
+		return ctrl.Result{}, fmt.Errorf("could not get cluster sizing configuration: %w", err)
+	}
+
+	if condition := meta.FindStatusCondition(config.Status.Conditions, schedulingv1alpha1.ClusterSizingConfigurationValidType); condition == nil || condition.Status != metav1.ConditionTrue {
+		log.Info("Cluster sizing configuration is not valid, skipping for now")
+		return ctrl.Result{}, nil
+	}
+
+	sizeConfig := util.SizeConfiguration(&config, desiredSize)
+	if sizeConfig == nil {
+		return ctrl.Result{}, fmt.Errorf("could not find size configuration for size %s", desiredSize)
+	}
+
+	if err := r.updateAroHostedCluster(ctx, hc, desiredSize, &config); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Scheduler) updateAroHostedCluster(ctx context.Context, hc *hyperv1.HostedCluster, size string, config *schedulingv1alpha1.ClusterSizingConfiguration) error {
```

Comment:
Please let's keep this project as product agnostic building blocks as much as possible while let it be platform aware. E.g. replace UpdateARO... with UpdateAzure?
What's the distinction between updateAROHostedCluster and updateROSAHostedCluster? Please name the func according to what the func actually does and let the diff be a param if needed

Reply:
Also let's name the parent folders aws and azure consistently

Reply:
I had been using managed Azure previously - https://github.com/openshift/hypershift/pull/5417#discussion_r1920637513